File: 프로그래머스/0/181854. 배열의 길이에 따라 다른 연산하기/배열의 길이에 따라 다른 연산하기.py
def solution(arr, n):
    answer = []
    if len(arr) % 2 == 1:
        answer = [arr[i] + n if i %2 ==0 else arr[i] for i in range(len(arr))]
    else :
        answer = [arr[i] + n if i %2 ==1 else arr[i] for i in range(len(arr))]
    return answer

File: 프로그래머스/0/181854. 배열의 길이에 따라 다른 연산하기/배열의 길이에 따라 다른 연산하기.java
import java.util.*;
class Solution {
    List<Integer> rs = new ArrayList();
    public int[] solution(int[] arr, int n) {
        for (int i = 0; i < arr.length ; i++){
            if(arr.length % 2 == 1 && i % 2 == 0){
                rs.add(arr[i] + n);
            } else if (arr.length % 2 == 0 && i % 2 == 1){
                rs.add(arr[i] + n);
            }else{
                rs.add(arr[i]);
            }
        }
        
        return rs.stream().mapToInt(i->(int)i).toArray();
    }
}